lib/posthog/feature_flags.rb

```diff
@@ -4,35 +4,70 @@
 require 'posthog/version'
 require 'posthog/logging'
 require 'digest'
+
 class PostHog
+
+  class InconclusiveMatchError < StandardError
+  end
+
   class FeatureFlagsPoller
     include PostHog::Logging
 
     def initialize(polling_interval, personal_api_key, project_api_key, host)
-      @polling_interval = polling_interval || 60 * 5
+      @polling_interval = polling_interval || 30
       @personal_api_key = personal_api_key
       @project_api_key = project_api_key
-      @host = host || 'app.posthog.com'
+      @host = host
       @feature_flags = Concurrent::Array.new
+      @group_type_mapping = Concurrent::Hash.new
       @loaded_flags_successfully_once = Concurrent::AtomicBoolean.new
 
       @task =
         Concurrent::TimerTask.new(
           execution_interval: polling_interval,
-          timeout_interval: 15
         ) { _load_feature_flags }
 
-      # load once before timer
-      load_feature_flags
-      @task.execute
+      # If no personal API key, disable local evaluation & thus polling for definitions
+      if !@personal_api_key.nil?
+        # load once before timer
+        load_feature_flags
```

Comment:
Suggestion: let's provide a configuration option to disable feature flags.

Reply:
You mean, other than the implicit provision of personal api key?

Reply:
Exactly